### ISSUE TITLE
Mermaid diagrams cannot be centered anymore

### ISSUE THREAD

**Issue by mrcrilly** [squidfunk/mkdocs-material#2953]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Previously I was using some sneaky custom CSS to center the Mermaid diagrams on the page:

```css
 .mermaid-experimental {
    text-align: center;
 }
```
The `<div>` element that Mermaid diagrams are now placed inside of are completely classless, so I cannot target them with custom css at all. If I try to, I end up centering a lot of other elements on the screen.

### Expected behaviour

I would like the Mermaid diagrams to please have a class I can target in my custom CSS, allowing me to manipulate their position on the page.

### Actual behaviour

The `<div>` element wrapping Mermaid diagrams is classless and thus cannot be easily referenced in CSS code.

### Steps to reproduce

1. Create a Mermaid diagram
2. Try to reference it in CSS
3. You cannot

### Package versions

- Python: `3.8.10`
- MkDocs: `1.2.2`
- Material: `7.2.4+insiders.2.13.0`

### Configuration

```yaml
site_name: End-to-End DevOps
site_description: The End-to-End DevOps Book
site_author: Michael Crilly
copyright: Copyight 2021 OpsFactory Pty Ltd

extra_css:
  - css/extra.css
nav:
  - Start here: index.md

theme:
  extra:
  name: material
  features:
    - navigation.indexes
    - navigation.tabs
    - search.highlight
plugins:
  - search
  - minify:
      minify_html: true
  - git-revision-date-localized
  - macros
markdown_extensions:
  - pymdownx.superfences:
      custom_fences:
        - name: mermaid
          class: mermaid-experimental
          format: !!python/name:pymdownx.superfences.fence_code_format
  - toc:
      permalink: true
  - pymdownx.arithmatex
  - pymdownx.betterem:
      smart_enable: all
  - pymdownx.caret
  - pymdownx.critic
  - pymdownx.details
  - pymdownx.emoji:
      emoji_index: !!python/name:materialx.emoji.twemoji
      emoji_generator: !!python/name:pymdownx.emoji.to_svg
  - pymdownx.magiclink
  - pymdownx.mark
  - pymdownx.smartsymbols
  - pymdownx.tasklist:
      custom_checkbox: true
  - pymdownx.tilde
  - admonition
  - codehilite
  - footnotes
  - pymdownx.highlight:
      linenums: true
  - abbr
  - pymdownx.tabbed
```

### System information

- Operating system: Ubuntu Linux, Windows 10, macOS
- Browser: Firefox, Edge

**Comment by facelessuser** [comment]
Yeah, at the very least, it seems Mermaid diagrams should attach a class to the wrapper div as everything else is under a shadow DOM now.

**Comment by squidfunk** [comment]
Thanks for suggesting - a very reasonable request! I've added the class back in the latest commit on the Insiders repository. The `*-experimental` suffix was silently deprecated in a previous release, so users should just use `mermaid` from now on. It still works as a value in `mkdocs.yml` but will be dropped in 8.x, and [the documentation](https://squidfunk.github.io/mkdocs-material/reference/diagrams/?h=mermaid#superfences) was also updated.

I checked, and the following CSS now correctly centers diagrams that do not span the entire container:

``` css
.mermaid {
    text-align: center;
 }
```

**Comment by facelessuser** [comment]
@squidfunk can Material's auto Mermaid handling be disabled? I've been using custom integration with my own styling for a while, and for now, I'd like to not have the overhead of them both trying to run and possibly conflict.

**Comment by squidfunk** [comment]
@facelessuser the integration is only mounted when a code block with the class `mermaid` is seen. If you use another class for the custom fence, the logic is not executed.

**Comment by facelessuser** [comment]
@squidfunk I understand, I am using and have been using `mermaid` for a very long time and am looking to avoid issues. I mean, yeah, I can change the class I use everywhere, but I was hoping to avoid that.

**Comment by facelessuser** [comment]
I do understand from a Material perspective using `mermaid` is convenient, so I do understand if there is no reasonable way to disable diagrams, I may just be forced to use an alternate class and update my docs accordingly. I'm just not ready to give up custom styling and don't won't to deal with breakage when this goes live.

**Comment by squidfunk** [comment]
I understand the troubles this is causing you, but I'd say that yours is a special case because normally there is no need to use a custom fence with the same name when Material for MkDocs provides its own, native integration. There is no runtime overhead if a page contains no diagrams. Furthermore, there's still enough time available until this feature hits the community edition given the current growth rate, so if this causes trouble in "normal" use cases, I'd consider changing it.

**Comment by facelessuser** [comment]
Cool, I guess I'll migrate everything to something like `diagrams` . I test insiders now and then, but I'm not actively using it, so when it does drop, this will allow me to avoid surprise breakage. At some point, I may migrate to Material's implementation once it is public, I just want time to test before I do and don't want to be surprised when it does go public.

**Comment by squidfunk** [comment]
@facelessuser Sounds legit. @mrcrilly The fix was just released as part of 7.2.4-insiders-2.13.1.

**Comment by mrcrilly** [comment]
@squidfunk I'm already an insider, but where can I throw you/the project $100?
